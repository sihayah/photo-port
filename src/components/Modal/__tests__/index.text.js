import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Modal from '..';

const mockToggleModal = jest.fn();
const currentPhoto = {
    name: 'Pug smile',
    category: 'portraits',
    description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc ultricie',
    index: 1
  };

afterEach(cleanup);

describe('it renders', () => {
    it('renders', () => {
        render(<Modal 
            currentPhoto = {currentPhoto} 
            />);
    })
    it('Modal matches snapshot', () => {
        const { asFragment } = render(
            <Modal 
            currentPhoto = {currentPhoto} 
            /> 
        )
        expect(asFragment).toMatchSnapshot();
    })
})

describe('Click Event', () => {
    it('call onClose handler', () => {
        const { getByText } = render(<Modal 
            onClose={mockToggleModal}
            currentPhoto = {currentPhoto} 
            />)
        fireEvent.click(getByText('Close this modal'));
        expect(mockToggleModal).toHaveBeenCalledTimes(1);
    })
})