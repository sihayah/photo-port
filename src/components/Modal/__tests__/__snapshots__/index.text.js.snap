// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders Modal matches snapshot 1`] = `[Function]`;
